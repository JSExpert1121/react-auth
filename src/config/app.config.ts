export const KAKAOMAP_API_KEY = '44a509cf4bf161e250c940454df5977c';
export const SUPPORTED_COUNTRIES = ['kr', 'jp', 'cn', 'hk', 'sg', 'tw', 'th', 'my', 'au', 'us', 'gb', 'de', 'fr', 'kz', 'ae', 'in'];
// export const COUNTRY_CODES = ['82', '81', '86', '852', '65', '886', '66', '60', '61', '1', '44', '49', '49', '33', '7', '971', '91'];
