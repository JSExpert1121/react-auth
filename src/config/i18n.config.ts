/**
 * @name i18n.config
 * @description config file for i18n. currently support ko and en
 * @version 1.0
 * @todo add locales
 * @author Darko
 */
const enMessages = require('translations/en.json');
const koMessages = require('translations/ko.json');

export const DEFAULT_LOCALE = 'en';

export const appLocales = [
    { code: 'en', name: 'English' },
    { code: 'ko', name: 'Korean' }
];

// auxilliary function to generate messages for each locales
export const formatMessages: (locale: string, messages: Record<string, string>) => Record<string, string> = (locale, messages) => {
    const defaultFormattedMessages =
        locale !== DEFAULT_LOCALE
            ? formatMessages(DEFAULT_LOCALE, enMessages)
            : {};
    const flattenFormattedMessages = (formattedMessages: Record<string, string>, key: string) => {
        const formattedMessage =
            !messages[key] && locale !== DEFAULT_LOCALE
                ? defaultFormattedMessages[key]
                : messages[key];
        return Object.assign(formattedMessages, { [key]: formattedMessage });
    };
    return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

export const translationMessages = {
    en: formatMessages('en', enMessages),
    ko: formatMessages('ko', koMessages),
};
