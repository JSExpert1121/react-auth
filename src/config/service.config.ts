export const API_GATEWAY_ADDRESS = 'https://apig.t.moria.hitit.xyz:8765';
export const OAUTH_CLIENT_ID = 'web';
export const OAUTH_CLIENT_SECRET = 'hititweb42!';