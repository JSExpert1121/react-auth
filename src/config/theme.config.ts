import { createMuiTheme } from '@material-ui/core/styles';

const theme = createMuiTheme({
	palette: {
		primary: {
			main: '#3f51b5',
			dark: '#303f9f',
		},
		secondary: {
			light: '#0033EE',
			main: '#1100BB',
			contrastText: '#000000',
		},
		background: {
			default: '#FFF'
		}
	},
	typography: {
		fontFamily: "\"Noto Sans\", \"Noto Sans KR\", sans-serif"
	},
	overrides: {
		MuiAppBar: {
			colorPrimary: {
				color: '#111',
				backgroundColor: 'white'
			}
		},
		MuiContainer: {
			root: {
				padding: 0
			}
		},
		MuiBadge: {
			colorSecondary: {
				color: 'white'
			}
		},
		MuiExpansionPanel: {
			root: {
				margin: '8px 0',
				'&:before': {
					height: 0
				}
			}
		},
		MuiExpansionPanelSummary: {
			root: {
				padding: 0,
				'&:before': {
					height: 0
				}
			},
			content: {
				margin: 0,
				flexGrow: 0,
				'&.Mui-expanded': {
					margin: 0
				}
			}
		},
		MuiExpansionPanelDetails: {
			root: {
				padding: 8
			}
		}
	}
});

export default theme;
