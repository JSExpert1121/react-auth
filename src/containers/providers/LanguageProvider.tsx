import React from 'react';
import { connect } from 'react-redux';
import { IntlProvider } from 'react-intl/';

export interface ILanguageProviderProps {
    locale: string;
    messages: Record<string, Record<string, string>>;
}

export const LanguageProvider: React.FC<ILanguageProviderProps> = (props) => {
    return (
        <IntlProvider
            locale={props.locale}
            key={props.locale}
            messages={props.messages[props.locale]}
        >
            {props.children}
        </IntlProvider>
    );
}

const mapStateToProps = (state: any) => ({
    locale: state.global.locale
});

export default connect(mapStateToProps)(LanguageProvider);