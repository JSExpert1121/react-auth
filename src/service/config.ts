import { oauthClient } from './base';

export const setLanguage = (lang: string) => {
    oauthClient.setLanguage(lang);
}
