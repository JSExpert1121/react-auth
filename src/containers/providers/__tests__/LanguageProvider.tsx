import React from 'react';
import { Route, Redirect } from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';

import { createShallow } from '@material-ui/core/test-utils';
import { LanguageProvider } from '../LanguageProvider';
import { translationMessages } from 'config/i18n.config';
import { IntlProvider } from 'react-intl';

configure({ adapter: new Adapter() });

describe('Language Provider', () => {

    const shallow = createShallow();
    const wrapper = shallow(<LanguageProvider locale='en' messages={translationMessages} />);

    test('snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });

    test('children', () => {
        expect(wrapper.find(IntlProvider)).toHaveLength(1);
        expect(wrapper.props().children).toEqual(wrapper.find(IntlProvider).props().children);
    })

});