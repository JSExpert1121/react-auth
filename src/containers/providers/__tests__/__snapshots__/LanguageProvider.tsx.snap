// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Provider snapshot 1`] = `ShallowWrapper {}`;
